pkg/controllers/grafana-operator/controller.go

```diff
@@ -400,3 +480,60 @@ func singleResourceInformer(name string, namespace string, resource string, obje
 		cache.Indexers{},
 	)
 }
+
+func creationResult(d time.Duration, err error) reconcileResult {
```

Comment:
I understand what this is trying to do, but is it really in the scope of reducing the error rate of the operator? I agree that we don't want extra reconcile runs, but to me this looks more like an optimization. I am also not sure whether this is the correct way to solve the problem, or if we should look into using predicates.

Reply:
> is it really in the scope of reducing the error rate of the operator?
yes, otherwise what happens is that the `Get` call says object not found (since the cache is invalid) but the object actually exists.

Reply:
Which `get` call exactly? Can we handle just that case?